src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -830,6 +832,67 @@ public function isAbstract()
     }
 
     /**
+     * Whether this definition is deprecated, that means it should not be called
+     * anymore.
+     *
+     * @param bool   $status
+     * @param string $template Template message to use if the definition is deprecated
+     *
+     * @return Definition the current instance
+     *
+     * @throws InvalidArgumentException In case the template message contains invalid chars such as \r, \n or end of long comments.
+     * @throws InvalidArgumentException In case the template message does not contain required %service_id% variable.
+     *
+     * @api
+     */
+    public function setDeprecated($status = true, $template = null)
+    {
+        $this->deprecated = (bool) $status;
+
+        if (null !== $template) {
+            if (preg_match('#[\r\n]|\*/#', $template)) {
+                throw new InvalidArgumentException('Invalid characters found in deprecation message');
+            }
+
+            if (false === strpos($template, '%service_id%')) {
+                throw new InvalidArgumentException('The deprecation message *must* contain %service_id% variable');
+            }
+
+            $this->deprecationTemplate = $template;
+        }
+
+
+        return $this;
+    }
+
+    /**
+     * Whether this definition is deprecated, that means it should not be called
+     * anymore.
```

Comment:
On one line, I suggest using a shorter one: `Whether this definition is deprecated or not.`